src/datachain/lib/dc.py

```diff
@@ -331,22 +346,80 @@ def from_storage(
             chain = DataChain.from_storage("s3://my-bucket/my-dir")
             ```
         """
-        func = get_file(type)
-        return (
-            cls(
-                path,
+        file_type = get_file_type(type)
+
+        if client_config is None:
+            client_config = {}
+
+        if anon:
+            client_config["anon"] = True
+
+        session = Session.get(
+            session,
+            catalog=kwargs.get("catalog"),
+            client_config=client_config,
+            in_memory=in_memory,
+        )
+
+        client, path = Client.parse_url(
+            uri, session.catalog.cache, **session.catalog.client_config
+        )
+
+        # clean path without globs
+        lst_path = (
+            posixpath.dirname(path)
+            if glob.has_magic(os.path.basename(os.path.normpath(path)))
+            or client.fs.isfile(uri)
+            else path
+        )
+        lst_uri = f"{client.uri}/{lst_path.lstrip('/')}"
+
+        ds_name = listing_dataset_name(client.uri, posixpath.join(lst_path, ""))
+
+        for ds in cls.datasets(
```

Comment:
does it return all datasets + listsings? can we return _only_ listings?

otherwise, code like `listing_expired(ds.created_at)` below look a bit strange if it's applied to a regular datasets

Reply:
Yes, this will return datasets + listings (special kind of datasets). 
In followup [PR](https://github.com/iterative/datachain/pull/331) after this one I've added `def listings()`  [method](https://github.com/iterative/datachain/pull/331/files#diff-0b1cfd3d89c057cc57811e07eb20298eeaaee60c54cc66e5128f618d144f42eeR597-R633) that returns only listing datasets which I will add to be used here instead of `cls.datasets(...)`